### PR TITLE
net_plugin transaction dedup cache handling

### PR DESCRIPTION
**Pull request by heifner**
- Add option `p2p-dedup-cache-expire-time-sec` for configurating the transaction deduplication cache expiration time.
- Simplify transaction deduplication cache handling to just be a configurable expire time.
  - The dedup cache, `local_txns`, is simply an optimization in net_plugin to avoid tripping duplication transaction exception in `controller`.
  - The current implementation (before this PR) prevents clients from re-trying transactions as the `net_plugin` would not accept the transaction again until it expired or it appeared in a block.

### CODE REVIEW

File: plugins/net_plugin/net_plugin.cpp [hunk 1]
```diff
@@ -2071,17 +2029,13 @@ namespace eosio {
    }
 
    void dispatch_manager::bcast_transaction(const packed_transaction& trx) {
-      const auto& id = trx.id();
-      time_point_sec trx_expiration = trx.expiration();
-      node_transaction_state nts = {id, trx_expiration, 0, 0};
-
       std::shared_ptr<std::vector<char>> send_buffer;
-      for_each_connection( [this, &trx, &nts, &send_buffer]( auto& cp ) {
+      auto now = fc::time_point::now();
```
Comment (brianjohnson5972) [inline plugins/net_plugin/net_plugin.cpp:2033]: make this const to make it clear that now is the same for all connections.

Comment (brianjohnson5972) [inline plugins/net_plugin/net_plugin.cpp:2033]: make this const to make it clear that now is the same for all connections.

**Review by brianjohnson5972 — CHANGES_REQUESTED** [review]
2 minor changes

**Comment by heifner** [comment]
> 2 minor changes

I only see one requested change.

File: plugins/net_plugin/net_plugin.cpp (hunk 1 above)
Comment (heifner) [reply plugins/net_plugin/net_plugin.cpp]: Done

**Review by brianjohnson5972 — APPROVED** [review]

**Pull request merged**